### DISCUSSION TITLE
Multiple line selection up or down

### DISCUSSION THREAD

**Discussion by chriskerluke** [vscode-neovim/vscode-neovim#1461] [Q&A] [answered]
I am trying to set up some key mappings, I like using visual block mode and moving multiple lines up or down.

The visual selection seems stay in the same spot or grow, but then the lines start wrapping instead of moving up/down.
Is there something I need to do to synchronise or keep the selection on the moving text?

On a more general note is this a vim anti pattern and I should be using something else to move multiple lines up?
Just selection dd and P?

**Comment by theol0403** [comment]
Show your bindings. Are you moving the lines from vscode or neovim side? Their performance will differ.

**Reply by chriskerluke** [reply]
I have tried both from VScode and from neovim, currently I am sending Alt+J  Alt+k to neovim

and trying this
`map("n", "\<A-k\>", "<Cmd>call VSCodeNotify('editor.action.moveLinesUpAction')<CR>", { desc ="move line down"}) -- move line up(n)
map("n", "\<A-j\>", "<Cmd>call VSCodeNotify('editor.action.moveLinesDownAction')<CR>", { desc ="move line up"}) -- move line down(n)

map("v", "\<A-j\>", '<cmd>call VSCodeNotify("editor.action.moveLinesDownAction")<CR>', { desc ="movae line up"}) -- move line down(n)SS
map("v", "\<A-k\>", '<cmd>call VSCodeNotify("editor.action.moveLinesDownAction")<CR>', { desc ="move line down"}) -- move line up(n)`

Should I be doing this in vscode for better performance?

**Reply by theol0403** [reply, marked as answer]
No, I mean moving the lines from the neovim side. There are a lot of pure-neovim plugins/bindings that do this for you and should work better.

**Reply by chriskerluke** [reply]
So I should be looking at using a neovim plugin to facilitate this command/motion?

I thought maybe I was screwing up the selection as it seems to changed during the commands as well.

But this didn't work either.
`map("v", "<A-j>", '<cmd>call VSCodeNotifyRange("editor.action.moveLinesDownAction",line("v"), line("."), 1)<CR>', { desc ="movae line up"}) -- move line down(n)SS
map("v", "<A-k>", '<cmd>call VSCodeNotifyRange("editor.action.moveLinesDownAction",line("v"), line("."), 1)<CR>', { desc ="move line down"}) -- move line up(n)`

**Reply by theol0403** [reply]
Those are all examples of doing it from the vscode side, it doesn't matter whether its triggered from a vscode or neovim binding

**Comment by chriskerluke** [comment]
This is what I am talking about, am I mangling the selection somehow?
![Recording 2023-09-17 at 14 48 45](https://github.com/vscode-neovim/vscode-neovim/assets/97127657/b8d030b1-e27e-4eaf-b409-8ffefceba2eb)

But I'll try and find a plugin to use purely through neovim.

**Reply by theol0403** [reply]
I'm not sure if you're doing something wrong or not but it is likely that neovim just doesnt know how to handle the buffer update events from vscode and mantain a correct selection.

**Reply by chriskerluke** [reply]
Thanks for the help, I found a plugin to facilitate this as per your suggestions

**Reply by shivamchandra75** [reply]
> Thanks for the help, I found a plugin to facilitate this as per your suggestions

@chriskerluke  please share the plugin you found to solve the problem...

**Reply by amikitevich** [reply]
https://github.com/fedepujol/move.nvim plugin works for me

```lua
return {
  "fedepujol/move.nvim",
  vscode = true,
  opts = {},
  keys = {
    { "J", ":MoveBlock(1)<CR>", mode = "v", silent = true, desc = "Move block down" },
    { "K", ":MoveBlock(-1)<CR>", mode = "v", silent = true, desc = "Move block up" },
  },
}
```
